controllers/operands/kubevirt.go

```diff
@@ -246,13 +250,15 @@ func (*kubevirtHooks) updateCr(req *common.HcoRequest, Client client.Client, exi
 		return false, false, errors.New("can't convert to KubeVirt")
 	}
 	if !reflect.DeepEqual(found.Spec, virt.Spec) ||
-		!reflect.DeepEqual(found.Labels, virt.Labels) {
+		!reflect.DeepEqual(found.Labels, virt.Labels) ||
+		!reflect.DeepEqual(found.Annotations, virt.Annotations) {
 		if req.HCOTriggered {
 			req.Logger.Info("Updating existing KubeVirt's Spec to new opinionated values")
 		} else {
 			req.Logger.Info("Reconciling an externally updated KubeVirt's Spec to its opinionated values")
 		}
 		hcoutil.DeepCopyLabels(&virt.ObjectMeta, &found.ObjectMeta)
+		hcoutil.DeepCopyAnnotations(&virt.ObjectMeta, &found.ObjectMeta)
 		virt.Spec.DeepCopyInto(&found.Spec)
 		err := Client.Update(req.Ctx, found)
 		if err != nil {
```

Comment:
Maybe something like this?

(Please notice that the suggested code is longer than the original code. this is because of github limitations)
```suggestion
	if outOfBound := !reflect.DeepEqual(found.Spec, virt.Spec) ||
		!reflect.DeepEqual(found.Labels, virt.Labels); 
		outOfBound || !reflect.DeepEqual(found.Annotations, virt.Annotations) {
		if req.HCOTriggered {
			req.Logger.Info("Updating existing KubeVirt's Spec to new opinionated values")
		} else {
			req.Logger.Info("Reconciling an externally updated KubeVirt's Spec to its opinionated values")
		}
		hcoutil.DeepCopyLabels(&virt.ObjectMeta, &found.ObjectMeta)
		hcoutil.DeepCopyAnnotations(&virt.ObjectMeta, &found.ObjectMeta)
		virt.Spec.DeepCopyInto(&found.Spec)
		err := Client.Update(req.Ctx, found)
		if err != nil {
            			return false, false, err
		}

        	overwritten = outOfBound && !req.HCOTriggered
		return true, overwritten, nil
	}
	return false, false, nil

```

Reply:
After discussing offline - I adopted this change in a separate commit, so that it's well documented (Changed naming a bit - please tell me what you think).